test/master-playlist-controller.test.js

```diff
@@ -3582,6 +3585,80 @@ QUnit.test('creates source buffers after second trackinfo if demuxed', function(
 
 });
 
+QUnit.test('Uses audio codec from audio playlist for demuxed content', function(assert) {
+  const done = assert.async();
+
+  window.MediaSource.isTypeSupported = (type) => (/(mp4a|avc1)/).test(type);
+
+  this.requests.length = 0;
+  this.player.dispose();
+  this.player = createPlayer();
+  this.player.src({
+    src: 'manifest/dash-demuxed.mpd',
+    type: 'application/dash+xml'
+  });
+  this.clock.tick(1);
+
+  const createSourceBufferCalls = [];
+  const mpc = this.player.vhs.masterPlaylistController_;
+  const origCreateSourceBuffers =
+    mpc.sourceUpdater_.createSourceBuffers.bind(mpc.sourceUpdater_);
+
+  mpc.sourceUpdater_.createSourceBuffers = (codecs) => {
+    createSourceBufferCalls.push(codecs);
+    origCreateSourceBuffers(codecs);
+  };
+
+  openMediaSource(this.player, this.clock);
+
+  // master
+  this.standardXHRResponse(this.requests.shift());
+
+  assert.equal(createSourceBufferCalls.length, 0, 'have not created source buffers yet');
+
+  let trackinfo = 0;
+
+  const onTrackInfo = function() {
+    trackinfo++;
+    if (trackinfo !== 2) {
+      return;
+    }
+    assert.equal(createSourceBufferCalls.length, 1, 'called to create source buffers');
+    assert.deepEqual(
+      createSourceBufferCalls[0],
+      {
+        video: 'avc1.foo.bar',
+        audio: 'mp4a.foo.bar'
+      },
+      'passed default codecs'
+    );
+
+    const playlists = mpc.master().playlists;
+
+    assert.equal(typeof playlists[0].excludeUntil, 'undefined', 'did not blacklist the first playlist');
```

Comment:
Might be nice to verify the codecs of each playlist or to specify in a comment why they are being blacklisted.